### ISSUE TITLE
[inductor][cpu]LayoutLMForSequenceClassification AMP single threadstatic/dynamic shape default/CPP wrapper accuracy failure

### ISSUE THREAD

**Issue by zxd1997066** [pytorch/pytorch#128510]
### 🐛 Describe the bug

ffc202a1b91def8c81a6eb9a39777bc7e149e1ee
```
/workspace/pytorch# bash inductor_single_run.sh single inference accuracy huggingface LayoutLMForSequenceClassification amp first static default
Testing with freezing on.
single-thread testing....
/opt/conda/lib/python3.8/site-packages/huggingface_hub/file_download.py:1132: FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.
  warnings.warn(
loading model: 0it [00:01, ?it/s]
cpu  eval  LayoutLMForSequenceClassification
WARNING:common:Trying to call the empty_gpu_cache for device: cpu, which is not in list [cuda, xpu]
WARNING:common:Trying to call the empty_gpu_cache for device: cpu, which is not in list [cuda, xpu]
WARNING:common:Trying to call the empty_gpu_cache for device: cpu, which is not in list [cuda, xpu]
E0612 14:35:01.137710 128914524942784 torch/_dynamo/utils.py:1476] RMSE (res-fp64): 0.00157, (ref-fp64): 0.00046 and shape=torch.Size([1, 2]). res.dtype: torch.bfloat16, multiplier: 3.000000, tol: 0.001000
E0612 14:35:01.137950 128914524942784 torch/_dynamo/utils.py:1390] Accuracy failed for key name logits
fail_accuracy
WARNING:common:Trying to call the empty_gpu_cache for device: cpu, which is not in list [cuda, xpu]
dev,name,batch_size,accuracy,calls_captured,unique_graphs,graph_breaks,unique_graph_breaks,autograd_captures,autograd_compiles,cudagraph_skips
cpu,LayoutLMForSequenceClassification,1,fail_accuracy,396,3,2,1,0,0,2
```
c4468518293f524d7ff1b2514d59753870dc232c
```
/workspace/pytorch# bash inductor_single_run.sh single inference accuracy huggingface LayoutLMForSequenceClassification amp first static default
Testing with freezing on.
single-thread testing....
/opt/conda/lib/python3.8/site-packages/huggingface_hub/file_download.py:1132: FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.
  warnings.warn(
loading model: 0it [00:01, ?it/s]
cpu  eval  LayoutLMForSequenceClassification
WARNING:common:Trying to call the empty_gpu_cache for device: cpu, which is not in list [cuda, xpu]
WARNING:common:Trying to call the empty_gpu_cache for device: cpu, which is not in list [cuda, xpu]
WARNING:common:Trying to call the empty_gpu_cache for device: cpu, which is not in list [cuda, xpu]
pass
WARNING:common:Trying to call the empty_gpu_cache for device: cpu, which is not in list [cuda, xpu]
dev,name,batch_size,accuracy,calls_captured,unique_graphs,graph_breaks,unique_graph_breaks,autograd_captures,autograd_compiles,cudagraph_skips
cpu,LayoutLMForSequenceClassification,1,pass,396,3,2,1,0,0,2
```

### Versions

</table><p>SW info</p><table border="1" class="dataframe table">
  <thead>
    <tr style="text-align: right;">
      <th>name</th>
      <th>target_branch</th>
      <th>target_commit</th>
      <th>refer_branch</th>
      <th>refer_commit</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>torchbench</td>
      <td>main</td>
      <td>d6015d42</td>
      <td>main</td>
      <td>d6015d42</td>
    </tr>
    <tr>
      <td>torch</td>
      <td>main</td>
      <td>f681e3689b857b8811f19d60d439bfb3fb2dd2d3</td>
      <td>main</td>
      <td>c19ad112f65ab88af3eeb319dda0675c076b0b67</td>
    </tr>
    <tr>
      <td>torchvision</td>
      <td>main</td>
      <td>0.19.0a0+d23a6e1</td>
      <td>main</td>
      <td>0.19.0a0+d23a6e1</td>
    </tr>
    <tr>
      <td>torchtext</td>
      <td>main</td>
      <td>0.16.0a0+b0ebddc</td>
      <td>main</td>
      <td>0.16.0a0+b0ebddc</td>
    </tr>
    <tr>
      <td>torchaudio</td>
      <td>main</td>
      <td>2.2.0a0+1980f8a</td>
      <td>main</td>
      <td>2.2.0a0+1980f8a</td>
    </tr>
    <tr>
      <td>torchdata</td>
      <td>main</td>
      <td>0.7.1a0+0790338</td>
      <td>main</td>
      <td>0.7.1a0+0790338</td>
    </tr>
    <tr>
      <td>dynamo_benchmarks</td>
      <td>main</td>
      <td>nightly</td>
      <td>main</td>
      <td>nightly</td>
    </tr>
  </tbody>
</table>

</table>

Repro:
[inductor_single_run.sh](https://github.com/chuanqi129/inductor-tools/blob/main/scripts/modelbench/inductor_single_run.sh)
bash inductor_single_run.sh single inference accuracy huggingface LayoutLMForSequenceClassification amp first dynamic/static default/cpp
Suspected guilty commit: https://github.com/pytorch/pytorch/commit/ffc202a1b91def8c81a6eb9a39777bc7e149e1ee
[huggingface-LayoutLMForSequenceClassification-inference-amp-static-default-single-accuracy-crash_guilty_commit.log](https://github.com/user-attachments/files/15806944/huggingface-LayoutLMForSequenceClassification-inference-amp-static-default-single-accuracy-crash_guilty_commit.log)
cc @ezyang @anijain2305 @chauhang @WeizhuoZhang-intel @chuanqi129

**Comment by chuanqi129** [comment]
Looks live we need to land the PR #127758 ASAP

**Comment by zxd1997066** [comment]
Hi @Chillee , according to the bisect search log and test results, the PR https://github.com/pytorch/pytorch/pull/124451 may introduce this accuracy failure on CPU, could you please help to double check it?

**Comment by Chillee** [comment]
Is it a real accuracy issue?

**Comment by leslie-fang-intel** [comment]
It may relate to the single thread optimization of customized SDPA kernel. @Valentine233 could you help to take a look?

**Comment by Valentine233** [comment]
Reproduced ONLY for amp + single thread.

The guilty commit makes the model hit the SDPA pattern 1. The accuracy fails even when going into the math-version SDPA.

**Comment by Chillee** [comment]
It sounds like this is actually a good optimization then, since the PR leads to SDPA being matched when it previously wasn't :)
